hayeon/programmers-1/과일-장수.js

```diff
@@ -0,0 +1,30 @@
+function solution(k, m, score) {
+  const sortScore = score.sort((a, b) => b - a);
+
+  let result = 0;
+  let lowestScore = m - 1;
+
+  for (let i = 0; i < Math.floor(score.length / m); i++) {
+    result += score[lowestScore] * m;
+
+    lowestScore += m;
+  }
+  return result;
+}
```

Comment:
```suggestion
function solution(k, m, score) {
  const sortedScore = [...score].sort((a, b) => b - a);
  let result = 0;
  let lowestScore = m - 1;

  for (let i = 0; i < Math.floor(score.length / m); i++) {
    result += sortedScore[lowestScore] * m;

    lowestScore += m;
  }
  return result;
}
```
- 풀이 잘봤습니다. sortScore 변수를 선언해주셨는데 막상 sortScore를 사용하는 곳은 없네요 😂 sort가 원본 배열을 바꾼다는 점을 의도하신걸까요?? 그렇다기에는 sortScore 변수를 또 선언하신 이유가 궁금하네요 🤔
- sort메서드는 원본 배열을 바꿔주는 즉 사이드 이펙트가 있는 메서드기 때문에 새로운 변수에 담고 싶다라고 하시면 전개 연산자로 복사 후에 호출하셔야 합니다.

Reply:
헉... 그렇군요 sort사용할 때 주의해서 사용하겠습니다!!!.. 새로운 변수에 담아서 콘솔창에만 찍고 정작 사용한데가 없었네유,,,🤦‍♀️